src/test/compile-fail/issue-21131.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+macro_rules! a {
+    ( $b:ident $c:d $e:expr ) => () //~ ERROR: unrecognized builtin nonterminal d
```

Comment:
It would be great to have a more helpful error message, or perhaps a span_help suggesting some legal non-terminals or something